native/vendor/google/play/PlayTask.cpp

```diff
@@ -211,7 +211,54 @@ void PlayTask::callJSfuncWithJObject(se::Object* listener, const char* functionN
             recallAccess->_hashCode = callIntMethod(env, objClass, jobj, "hashCode");
             recallAccess->_sessionId = callStringMethod(env, objClass, jobj, "getSessionId");
             cc::callJSfunc(listener, functionName, recallAccess);
-        } else if(name == "java.lang.Integer") {
+        } else if(name == "com.google.android.gms.games.AnnotatedData") {
+            auto* annotatedData = new AnnotatedData;
+            annotatedData->_isStale = callBooleanMethod(env,  objClass, jobj, "isStale");
+            jmethodID methodId = env->GetMethodID(objClass, "get", "()Ljava/lang/Object;");
+            jobject achievementBufferObj = env->CallObjectMethod(jobj, methodId);
+            if (achievementBufferObj != nullptr) {
+                auto& achievementBuffer = annotatedData->_achievementBuffer;
+                jclass achievementBufferObjClass = env->GetObjectClass(achievementBufferObj);
+                int count = callIntMethod(env,  achievementBufferObjClass, achievementBufferObj, "getCount");
+                jmethodID methodId = env->GetMethodID(achievementBufferObjClass, "get", "(I)Ljava/lang/Object;");
+                for(int i = 0; i < count; ++i) {
+                    jobject achievementObj = env->CallObjectMethod(achievementBufferObj, methodId, i);
+                    jclass achievementObjClass = env->GetObjectClass(achievementObj);
+                    if (achievementObj != nullptr) {
+                        auto* achievement = new Achievement;

```

Comment:
How about adding a method for `new Achievement` in AchievementBuffer ? 
Currently, `new Achievement` is outside `AchievementBuffer` but `delete it` is in the destructor of AchievementBuffer.

Perhaps, adding a method:

```c++
class AchievementBuffer {
    ~AchievementBuffer() {
        for(auto* achievement : _achievements) {
            delete achievement;  // -->  Delete elements in AchievementBuffer class
        }
        _achievements.clear();
    }
//    ...
    Achievement* createAchievement() {
          auto* achievement = new Achievement(); // --> new elements also in AchievementBuffer class
          _achievements.push_back(achievement);
    }
};
```

And no need to add `friend class PlayTask;` for AchievementBuffer.

Reply:
This interface is not intended to be exported for use by ts.